File: GMT/GMT/Utility/checkParamFields.m
% function checkParamFields(param, requiredFields)
% Check whether a number of required fields are contained in parameter
% struct or object. Throws an error if required fields are missing.
%
% INPUT:
%    param - struct or object
%    requiredFields - cell array of strings containing required field names
% 
% Change log:
% Apr. 2012, M.Milczynki - created
% 30/04/2012, P.Hehrmann - added support for objects
function ok = checkParamFields(param, requiredFields)

if (isstruct(param))    % param is a struct
    for j = 1:length(requiredFields)
        if ~isfield(param, requiredFields{j})
            error('Wrong parameter names or paramter missing.');
        end
    end
    
elseif (isobject(param)) % param is an object
    for j = 1:length(requiredFields)  % attempt to access all required fields: throws error if not possible
        foo = param.(requiredFields{j});
    end
    
else % param is something different
    error('Argument ''param'' needs to be either a struct or an object.');
end

ok = 1;

File: GpyT/GpyT/Utility/checkParamFields.py
# -*- coding: utf-8 -*-
"""
Created on Fri Sep  6 10:20:48 2019

@author: beimx004
"""

def checkParamFields(par,requiredFields):
    if isinstance(par,dict):
        for field in requiredFields:
            if not field in par.keys():
                raise ValueError('Wrong parameter names or parameter missing.')
    elif isinstance(par,object):
        for field in requiredFields:
            foo = par.field;
    else:
        raise TypeError('Input Argument ''par'' needs to be a dict or an object.')
        
            